transactions/rbf.ts

```diff
@@ -0,0 +1,435 @@
+import { base64, hex } from '@scure/base';
+import * as btc from '@scure/btc-signer';
+import * as bip39 from 'bip39';
+import EsploraProvider from '../api/esplora/esploraAPiProvider';
+import { signLedgerPSBT } from '../ledger/psbt';
+import { Transport } from '../ledger/types';
+import SeedVault from '../seedVault';
+import { AccountType, NetworkType } from '../types';
+import { RecommendedFeeResponse, Transaction, UTXO } from '../types/api/esplora';
+import { bip32 } from '../utils/bip32';
+import { getBitcoinDerivationPath, getTaprootDerivationPath } from '../wallet';
+
+// TODO: A lot of the below can be done much more easily with the consolidation logic
+// TODO: so we should refactor this file to use that once merged
+
+const areByteArraysEqual = (a: undefined | Uint8Array, b: undefined | Uint8Array): boolean => {
+  if (!a || !b || a.length !== b.length) {
+    return false;
+  }
+
+  return a.every((v, i) => v === b[i]);
+};
+
+const getRbfTransactionSummary = (transaction: Transaction) => {
+  const transactionVSize = transaction.weight / 4;
+
+  const currentTransactionInputTotals = transaction.vin.reduce((total, input) => total + input.prevout.value, 0);
+  const currentTransactionOutputTotals = transaction.vout.reduce((total, output) => total + output.value, 0);
+
+  const currentFee = currentTransactionInputTotals - currentTransactionOutputTotals;
+  const currentFeeRate = Math.ceil(currentFee / transactionVSize);
+
+  const minimumRbfFee = Math.ceil(transaction.fee + transactionVSize);
+  const minimumRbfFeeRate = Math.ceil(minimumRbfFee / transactionVSize);
+
+  return { currentFee, currentFeeRate, minimumRbfFee, minimumRbfFeeRate };
+};
+
+const isTransactionRbfEnabled = (transaction: Transaction) => {
+  if (transaction.status.confirmed) {
+    return false;
+  }
+
+  return transaction.vin.some((input) => input.sequence < 0xffffffff - 1);
+};
+
+type RBFProps = {
+  btcAddress: string;
+  ordinalsAddress: string;
+  btcPublicKey: string;
+  ordinalsPublicKey: string;
+  seedVault: SeedVault;
+  addressIndex: number;
```

Comment:
Is this `addressIndex` the same property as `deviceAccountIndex` in our `Account` interface?https://github.com/secretkeylabs/xverse-core/blob/d1a9ef806d90b680dffaee8dbaa78430774cdcf7/types/account.ts#L3

Reply:
I think it's actually the `id` field on that interface.

I just checked in the Core library codebase and deviceAccountIndex doesn't seem to be used at all. Maybe it's something deprecated that we should clean up?

Reply:
It's used for ledger-related logic, we pass it from web-extension so I think that on the core side it's used under some different param name

Reply:
Should we rename the `addressIndex` here to `id` or `accountId` to make it more obvious?

Reply:
Yeah, that's a good idea. Will change it to addressIndex 👍

Reply:
Hmmm, I'm not sure. This is what the derivation path function looks like
<img width="591" alt="image" src="https://github.com/secretkeylabs/xverse-core/assets/3322667/7b34b20c-36f5-4fbc-be88-387531cf1f03">

I think maybe we are using account and address id and index in an inconsistent way 🤔 

I guess accountId is probably the best option with the current naming. Will change it to that 👍

Reply:
Though now we end up with this 😅 
<img width="369" alt="image" src="https://github.com/secretkeylabs/xverse-core/assets/3322667/a6c95baa-ef2e-4404-94bc-4721868475c1">

I'll keep it as account index and we can add a tech debt ticket to make naming more conventional 👍

Reply:
I just checked the code and yeah, basically the `addressIndex` is for the `deviceAccountIndex` property, it's used to distinguish ledger accounts when user has more than one

Reply:
But it seems confusing, so we should probably rename it

Reply:
Ah I see. Yeah, we'll need to rethink the naming with ledger and potentially other hw wallets around. I can see how this is actually confusing when we have the non-optional id field and the optional deviceAccountIndex field on the Account type.

I think you can see what makes the most sense in the arg type for RBF while you build with it and let me know what makes the most sense in terms of naming and structure and I'll put those changes in 👍